shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,310 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_platform_views_plugin.h"
+
+#include <gtk/gtk.h>
+#include <cstring>
+
+#include "flutter/shell/platform/linux/fl_gesture_helper.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_method_channel.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_standard_method_codec.h"
+
+static constexpr char kChannelName[] = "flutter/platform_views";
+static constexpr char kBadArgumentsError[] = "Bad Arguments";
+static constexpr char kCreateMethod[] = "create";
+static constexpr char kDisposeMethod[] = "dispose";
+static constexpr char kAcceptGestureMethod[] = "acceptGesture";
+static constexpr char kRejectGestureMethod[] = "rejectGesture";
+static constexpr char kEnterMethod[] = "enter";
+static constexpr char kExitMethod[] = "exit";
+
+struct _FlPlatformViewsPlugin {
+  GObject parent_instance;
+
+  FlMethodChannel* channel;
+
+  FlGestureHelper* gesture_helper;
+
+  GHashTable* factories;  // string -> FlPlatformViewFactory*
```

Comment:
As you know, `GHashTable` can destory value automatically when key is dropped. But for C++ `unordered_map`, to archive the same goal, we must create a custom deleter for `unique_ptr`, and explicitly annotate type when creating `unique_ptr`, which results in more codes.

Reply:
To avoid the code base being too confusing, we use GObject objects where possible. So we use GPtrArray instead of std::vector and GHashTable over std::map.